src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/ProjectSnapshotManagerInstantiator.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Composition;
+using OmniSharp.MSBuild.Notification;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin
+{
+    [Shared]
+    [Export(typeof(IMSBuildEventSink))]
+    internal class ProjectSnapshotManagerInstantiator : IMSBuildEventSink
+    {
+        // The entire purpose of this class is to ensure the project manager is instantiated.
```

Comment:
Could you expand this comment to include what layer will this be called and why we need it to be initialized at this layer?

Reply:
Sure